### PR TITLE
Up nrhm reports

### PR DESCRIPTION
**Pull request by lwyszomi**
@snopoke @czue 

I had to modify daterangepicker.js to change behavior described in case: http://manage.dimagi.com/default.asp?165775#923135

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 48 errors:

* custom/up_nrhm/reports/asha_reports.py, line 3 - F401 'ProjectReportParametersMixin' imported but unused
* custom/up_nrhm/static/up_nrhm/js/custom_daterangepicker.js, line 286 - Use '!==' to compare with '0'.
* custom/up_nrhm/static/up_nrhm/js/custom_daterangepicker.js, line 321 - Missing '()' invoking a constructor.
* custom/up_nrhm/static/up_nrhm/js/custom_daterangepicker.js, line 586 - Bad line breaking before '-'.
* custom/up_nrhm/static/up_nrhm/js/custom_daterangepicker.js, line 1244 - 'num' is already defined.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 296 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 297 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 393 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 397 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 399 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 613 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 709 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 776 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1245 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1252 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1260 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1269 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1274 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1282 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1348 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1350 - ['d'] is better written in dot notation.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1385 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1403 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1409 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1437 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1459 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1465 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1480 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1613 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1632 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 1823 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2021 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2151 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2169 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2349 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2470 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2497 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2509 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2575 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2579 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2584 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2589 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2594 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2599 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2604 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2611 - Use '===' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2654 - Use '!==' to compare with 'null'.
* custom/up_nrhm/static/up_nrhm/js/moment.js, line 2716 - Use '!==' to compare with 'null'.

File: custom/up_nrhm/static/up_nrhm/js/moment.js [hunk 1]
```diff
@@ -0,0 +1,3043 @@
...
+        lists = ['months', 'monthsShort', 'weekdays', 'weekdaysShort', 'weekdaysMin'],
+
+        updateInProgress = false;
+
+    // Pick the first defined of two or three arguments. dfl comes from
+    // default.
+    function dfl(a, b, c) {
+        switch (arguments.length) {
+            case 2: return a !== null ? a : b;
+            case 3: return a !== null ? a : b != null ? b : c;
```
Comment (dimagimon) [inline custom/up_nrhm/static/up_nrhm/js/moment.js]: Use '!==' to compare with 'null'.

**Comment by czue** [comment]
@lwyszomi sorry it's taken us so long to review this. I was interested in seeing whether there was a way to get this behavior that didn't require editing the datepicker source but didn't have a chance to look closer into that. curious how hard you tried to find that solution and what options didn't work.

@snopoke and I are both offline for the rest of the week so this probably won't get reviewed till next week.

**Comment by snopoke** [comment]
Chatted with Sri about this and decided to go with another option that doesn't require a custom date widget. See for context: http://manage.dimagi.com/default.asp?165775

File: custom/up_nrhm/filters.py [hunk 2]
```diff
@@ -104,16 +106,24 @@ def get_hierarchy(self):
...
+            ('sf3', 'Format-3 Block Consolidation of the functionality status'),
+            ('sf4', 'Format-4 Block Consolidation of the functionality status'),
+            ('sf5', 'Format-5 Functionality of ASHAs in blocks')
         ]
 
 
 class ASHAMonthFilter(MonthFilter):
     label = "Last month of the quarter"
+
+class NRHMDatespanFilter(DatespanFilter):
```
Comment (dimagimon) [inline custom/up_nrhm/filters.py]: E302 expected 2 blank lines, found 1

**Comment by lwyszomi** [comment]
@snopoke Updated, ready to review and merge.

**Pull request merged**